lib/types.js

```diff
@@ -1314,7 +1314,7 @@ class UnwrappedUnionType extends UnionType {
     }
     this._getIndex = _projectionFn
       ? generateProjectionIndexer(_projectionFn)
-      : generateDefaultIndexer(this.types);
+      : generateDefaultIndexer(this.types, this);
```

Comment:
Alternative is:

```suggestion
      : generateDefaultIndexer.bind(this)();
```

and then use `this` and `this.types` directly.

Reply:
Current is simpler.